tests/pyminisketch.py

```diff
@@ -0,0 +1,514 @@
+        ret.gf = self.gf
+        return ret
+
+    def merge(self, other):
+        """Merge a sketch with another sketch. Corresponds to XOR'ing their serializations."""
+        assert self.capacity == other.capacity
+        assert self.field_size == other.field_size
+        for i in range(self.capacity):
+            self.odd_syndromes[i] ^= other.odd_syndromes[i]
+
+    def decode(self, max_count=None):
+        """Decode the contents of this sketch.
+
+        Returns either a list of elements or None if undecodable.
+        """
+        # We know the odd syndromes s1=x+y+..., s3=x^3+y^3+..., s5=..., and reconstruct the even
+        # syndromes from this:
+        #  * s2 = x^2+y^2+.... = (x+y+...)^2 = s1^2
+        #  * s4 = x^4+y^4+.... = (x^2+y^2+...)^2 = s2^2
+        #  * s6 = x^6+y^6+.... = (x^3+y^3+...)^2 = s3^2
+        all_syndromes = [0 for _ in range(2 * len(self.odd_syndromes))]
+        for i in range(len(self.odd_syndromes)):
+            all_syndromes[i * 2] = self.odd_syndromes[i]
+            all_syndromes[i * 2 + 1] = self.gf.sqr(all_syndromes[i])
+        # Given the syndromes, find the polynomial that generates them.
+        poly = berlekamp_massey(all_syndromes, self.gf)
+        # Deal with failure and trivial cases.
+        if len(poly) == 0:
+            return None
+        if len(poly) == 1:
+            return []
+        if max_count is not None and len(poly) > 1 + max_count:
+            return None
+        # If the polynomial can be factored into (1-m1*x)*(1-m2*x)*...*(1-mn*x), then {m1,m2,...,mn}
+        # is our set. As each factor (1-m*x) has 1/m as root, we're really just looking for the
+        # inverses of the roots. We find these by reversing the order of the coefficients, and
+        # finding the roots.
+        roots = poly_find_roots(list(reversed(poly)), self.gf)
+        if len(roots) == 0:
+            return None
+        return roots
+
+class TestMinisketch(unittest.TestCase):
+    """Test class for Minisketch."""
+
+    @classmethod
+    def construct_data(cls, field_size, num_a_only, num_b_only, num_both):
+        """Construct two random lists of elements in [1..2**field_size-1].
+
+        Each list will have unique elements that don't appear in the other (num_a_only in the first
+        and num_b_only in the second), and num_both elements will appear in both."""
+        a_only, b_only, both = set(), set(), set()
+        while len(a_only) < num_a_only:
+            a_only.add(random.randrange(1, 1 << field_size))
+        while len(b_only) < num_b_only:
+            while True:
+                r = random.randrange(1, 1 << field_size)
+                if r not in a_only:
+                    break
+            b_only.add(r)
+        while len(both) < num_both:
+            while True:
+                r = random.randrange(1, 1 << field_size)
+                if r not in a_only and r not in b_only:
+                    break
+            both.add(r)
```

Comment:
I could claim that the reason is because random.sample doesn't work with ranges that are bigger than 2^63, but that'd be a lie; I just didn't know about random.sample. However it is still the case that it doesn't work.

I've changed the code to just simulate random.sample now, though.